metricbeat/module/linux/memory/data.go

```diff
@@ -129,21 +137,28 @@ func getHugePages(hostfs resolve.Resolver) (common.MapStr, error) {
 }
 
 // GetVMStat gets linux vmstat metrics
-func GetVMStat() (*sysinfotypes.VMStatInfo, error) {
-	// TODO: We may want to pull this code out of go-sysinfo.
-	// It's platform specific, and not used by anything else.
-	h, err := sysinfo.Host()
+func GetVMStat(hostfs resolve.Resolver) (map[string]uint64, error) {
+	vmstat_file := hostfs.ResolveHostFS("proc/vmstat")
+	content, err := ioutil.ReadFile(vmstat_file)
 	if err != nil {
-		return nil, errors.Wrap(err, "failed to read self process information")
-	}
-	vmstatHandle, ok := h.(sysinfotypes.VMStat)
-	if !ok {
-		return nil, errors.New("VMStat not available for platform")
+		return nil, errors.Wrapf(err, "error reading vmstat from %s", vmstat_file)
 	}
-	info, err := vmstatHandle.VMStat()
-	if err != nil {
-		return nil, errors.Wrap(err, "error getting VMStat info")
+
+	// I'm not a fan of throwing stuff directly to maps, but this is a huge amount of kernel/config specific metrics, and we're the only consumer of this for now.
+	vmstat := map[string]uint64{}
+	for _, metric := range strings.Split(string(content), "\n") {
+		parts := strings.SplitN(metric, " ", 2)
+		if len(parts) != 2 {
+			continue
+		}
+
+		num, err := strconv.ParseUint(string(parts[1]), 10, 64)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to parse value")
```

Comment:
[Question/Suggestion]

Do you think it would add some value to also report the key and value that could not be parsed?

Reply:
Ah, good idea.